website/docs/cluster-management/generic-clusters.mdx

```diff
@@ -0,0 +1,155 @@
+
+### Managing non-capi clusters
+
+Any kubernetes cluster whether capi or not can be added to Weave Gitops Enterprise.
+
+import TOCInline from "@theme/TOCInline";
+<TOCInline />;
+
+### How to create a kubeconfig secret using a service account
+
+1. Create a new service account on the remote cluster:
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: demo-01
+  namespace: default
+```
+
+2. Add RBAC permissions for the service account
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: impersonate-user-groups
+subjects:
+  - kind: ServiceAccount
+    name: demo-02
+    namespace: default
+roleRef:
+  kind: ClusterRole
+  name: user-groups-impersonator
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: user-groups-impersonator
+rules:
+  - apiGroups: [""]
+    resources: ["users", "groups"]
+    verbs: ["impersonate"]
+  - apiGroups: [""]
+    resources: ["namespaces"]
+    verbs: ["get", "list"]
+```
+
+This will allow WGE to introspect the cluster for available namespaces.
+
+Once we know what namespaces are available we can test whether the logged in user can access them via impersonation.
+
+3. Get the token of the service account
+
+First get the list of secrets of the service accounts by running the following command:
+
+```sh
+kubectl get secrets --field-selector type=kubernetes.io/service-account-token
+NAME                      TYPE                                  DATA   AGE
+default-token-lsjz4       kubernetes.io/service-account-token   3      13d
+demo-01-token-gqz7p       kubernetes.io/service-account-token   3      99m
+```
+
+`demo-01-token-gqz7p` is the secret that holds the token for `demo-01` service account
+
+To get the token of the service account run the following command:
+
+```sh
+TOKEN=$(kubectl get secret demo-01-token-gqz7p -o jsonpath={.data.token} | base64 -d)
+```
+
+4. Create a kubeconfig secret
+
+For the next step, the cluster certificate (CA) is needed. How you get hold of the certificate depends on the cluster. For GKE you can view it on the GCP Console: Cluster->Details->Endpoint->”Show cluster certificate”. You will need to copy the contents of the certificate into the `ca.crt` file used below.
+
+```sh
+CLUSTER_NAME=demo-01 \
+CA_CERTIFICATE=ca.crt \
+ENDPOINT=<control-plane-ip-address> \
+TOKEN=<token> ./tools/sa/static-kubeconfig.sh > demo-01-kubeconfig
+```
+
+Replace the following:
+
+- CLUSTER_NAME: the name of your cluster i.e. `demo-01`
+- ENDPOINT: the API server endpoint i.e. `34.218.72.31`
+- CA_CERTIFICATE: path to the CA certificate file of the cluster
+- TOKEN: the token of the service account retrieved in the previous step
+
+Finally create a secret for the generated kubeconfig:
+
+```sh
+kubectl create secret generic demo-01-kubeconfig \
+--from-file=value.yaml=./demo-01-kubeconfig \
+--dry-run=client -o yaml > demo-01-kubeconfig.yaml
+```
+
```

Comment:
HERE: Describe how to create a `GitopsCluster` object w/. secretRef and to commit to `./clusters/management/clusters/my-cluster.yaml`